src/mumble/ConnectDialog.cpp

```diff
@@ -592,13 +593,20 @@ void ServerItem::hideCheck() {
 			hide = (uiUsers == 0);
 	}
 	if (hide != ishidden) {
-		if (hide)
+		if (hide || !nameMatchesFilter())
 			siParent->removeChild(this);
 		else
 			siParent->addChild(this);
 	}
 }
 
+bool ServerItem::nameMatchesFilter()
+{
+	if (filter.isEmpty() || qsHostname.isEmpty())
```

Comment:
What does the hostname have to do with the filtering? :thinking:

Reply:
Countries and continents are `ServerItem`s too and would be filteres out. Since they have no hostname I use this to check if this `ServerItem` is a real Server or not.

Reply:
Maybe it would be worth it to extract the location related parts from that class?

Reply:
Ah okay that makes sense then.

> Maybe it would be worth it to extract the location related parts from that class?

Absolutely. But that's out of the scope of this PR (refactor is coming at some point)